lib/matplotlib/tests/test_axes.py

```diff
@@ -4871,6 +4871,17 @@ def test_set_get_ticklabels():
     ax[1].set_yticklabels(ax[0].get_yticklabels())
 
 
+@image_comparison(
+    baseline_images=['retain_tick_visibility'],
+    extensions=['png'],
+)
+def test_retain_tick_visibility():
+    fig, ax = plt.subplots()
+    plt.plot([0, 1, 2], [0, -1, 4])
+    plt.setp(ax.get_yticklabels(), visible=False)
+    ax.tick_params(axis="y", which="both", length=0)
```

Comment:
I think this test could be easily done without an image comparison by just checking the visibility of the ticks?

Reply:
The tricky bit is that one of the common ways in tests of checking wheter the ticks are visible was lying to you prior to this PR, (though if you went to the individual ticks, it would be correct) As such, I was looking to make sure it translated to the image generated.

Reply:
You can instead save to svg and use some long-ish strings as labels, and check that the labels don't appear as substrings in the svg file.